aten/src/ATen/native/ForeachUtils.h

```diff
@@ -88,7 +88,13 @@ bool has_same_attributes(Device expected_device, TensorList tensors) {
   return true;
 }
 
-bool will_promote_tensor(const Tensor& tensor, Scalar scalar) {
+bool will_promote_tensor(const Tensor& tensor, Scalar scalar, bool division_op = false) {
```

Comment:
Division is not the only op that has this problem. IIRC copysign also does int->float promotion. We should probably name the flag something to be more in line with what TensorIterator calls it: https://github.com/pytorch/pytorch/blob/ccd646696bad6a967b56c5097b59ed3da63ee28b/aten/src/ATen/TensorIterator.cpp#L843